proto/v3/message_contents/association.proto

```diff
@@ -0,0 +1,26 @@
+// Association types
+syntax = "proto3";
+
+package xmtp.v3.message_contents;
+
+option go_package = "github.com/xmtp/proto/v3/go/v3/message_contents";
+option java_package = "org.xmtp.proto.v3.message.contents";
+
+// Allows for us to update the format of the association text without
+// incrementing the entire proto
+enum AssociationTextVersion {
+    ASSOCIATION_TEXT_VERSION_UNSPECIFIED = 0;
+    ASSOCIATION_TEXT_VERSION_1 = 1;
+}
+
+// EIP191Association is used for all EIP 191 compliant wallet signatures
+message Eip191Association {
+    AssociationTextVersion association_text_version = 1;
```

Comment:
I'm only storing the `version` of the text and not the actual text signed. This should be enough information to reconstruct the text in the recipient (they already know the public key bytes and address from other fields in the message). Any differences in output will lead to an error verifying the signature.

Reply:
I think this could work for the Static Signature type in the short term, but this structure would limit the flexibility of the Association system. Namely in that there is an assumption of the association text which breaks the push towards self-describing associations. 

I get the push towards a functional E2E message flow so this makes sense today, but I think we'll need to take another look prior to release.

Reply:
That's true. I've been deliberately leaving other signature types out of my thinking, since I'm not sure which ones we'll need and when. 

I did make a oneof so we can add more concrete types as needed.